src/FormCheckInput.tsx

```diff
@@ -54,19 +60,45 @@ const FormCheckInput: BsPrefixRefForwardingComponent<
       type = 'checkbox',
       isValid = false,
       isInvalid = false,
+      indeterminate,
       // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
       as: Component = 'input',
+      checked,
+      onChange,
       ...props
     },
     ref,
   ) => {
     const { controlId } = useContext(FormContext);
     bsPrefix = useBootstrapPrefix(bsPrefix, 'form-check-input');
 
+    const [element, inputRef] = useCallbackRef<HTMLInputElement>();
+    const existingRef = useMergedRefs(inputRef, ref);
```

Comment:
i think using `useUncontrolledProp` on `checked` might improve readability - but I'm struggling to see how it would simplify things
```ts
    const changeHandler = (e) => {
      handleIndeterminate();
      onChange?.(e);
    };
    const [innerChecked, setInnerChecked] = useUncontrolledProp(
      checked,
      defaultChecked,
      changeHandler,
    );
```
we still would need get refs(to get `element.indeterminate`) & set `element.indeterminate` in `onChange` handler

Reply:
Yeah but you combine the call sites where that happens with one use effect that has `checked` as its dep trigger. 

It's not a big difference maybe

Reply:
idk whether `useUncontrolledProp` will work 
```ts
    const [innerChecked, setInnerChecked] = useUncontrolledProp(
      checked,
      defaultChecked,
      onChange,
    );
```
`onChange` expects the argument to be an `Event`, meanwhile we need to pass a boolean to the inner `useState` in `setInnerChecked`
https://github.com/jquense/uncontrollable/blob/dd40d92600566b0af14e66e0942b3066e2f62528/src/hook.tsx#L41

I've pushed a version with just a setState - if you know how to make it work with `useUncontrolledProp` let me know!